src/bacnet/basic/object/ai.c

```diff
@@ -537,7 +537,11 @@ void Analog_Input_Out_Of_Service_Set(uint32_t object_instance, bool value)
         if (pObject->Out_Of_Service != value) {
             pObject->Changed = true;
         }
-        pObject->Out_Of_Service = value;
+        if((pObject->Out_Of_Service = value)) {
```

Comment:
it's not clear whether this is just an assignment or a check of ==, please add a note to clarify.

Reply:
It is both :)

Reply:
Commented